### ISSUE TITLE
We can access the value in struct directly without converting it to Map in Stripe

### ISSUE THREAD

**Issue by kartikjagdale** [aviabird/gringotts#73]
In Stripe Gateway, in `card_params` and `address_params`  functions I think the struct is unnecessarily converted to map inorder to access the values. which could be easliy done using [dot] operator.
E.g: 
`card.name`
Please go through the link to see how to access values in struct.
[Get a value from a struct](https://elixir-examples.github.io/examples/get-a-value-from-a-struct)

 ```
defp card_params(%CreditCard{} = card) do
    card = Map.from_struct(card)

    [ "card[name]": card[:name],
      "card[number]": card[:number],
      "card[exp_year]": card[:year],
      "card[exp_month]": card[:month],
      "card[cvc]": card[:verification_code]
    ]   
  end
```

**Comment by zeus999** [comment]
card.name will return an error if `name` key is not present in the struct and everyone is not going to all of the values in the card params, so we have to make a check if a key is present or not, then only we can access the key with `dot operator`.

**Comment by kartikjagdale** [comment]
@zeus999 can we change this to the right thing please. `card.name` to `card.first_name` and `card.last_name`

However, I am still skeptical about converting the struct to map.
there might be a better way to do this. 
@ashish173, @oyeb please let me know what you think. :smiley:

**Comment by kartikjagdale** [comment]
@zeus999 
```
card.name will return an error if `name` key is not present in the struct and everyone is not going to pass all of the values in the card params, so we have to make a check if a key is present or not, then only we can access the key with `dot operator`.
```
Even if the key is not present in the struct, it will return a nil value and not a key error for if the user is not passing any key listed in struct but if some key that is not in the struct **defination**.    
  
and I think latter we can sanatize the params.

**Comment by zeus999** [comment]
@kartikjagdale let me check it once again and I am going to fix it as soon as possible.

**Comment by oyeb** [comment]
Fixed in #75 (6c56cfc)
